apps/src/templates/sessions/LockoutPanel.jsx

```diff
@@ -0,0 +1,280 @@
+          })}
+        </p>
+
+        {/* This field shows the current status of the validation. */}
+        {/* Parent Permission: Not Pending / Pending */}
+        <div style={styles.statusSection}>
+          <label
+            style={isRTL ? styles.statusLabelRTL : styles.statusLabel}
+            htmlFor="permission-status"
+          >
+            <strong>{i18n.sessionLockoutParentStatusField()}</strong>
+          </label>
+          <span
+            id="permission-status"
+            style={props.pendingEmail ? styles.pending : styles.notSubmitted}
+          >
+            <strong>
+              {props.pendingEmail
+                ? i18n.sessionLockoutStatusPending()
+                : i18n.sessionLockoutStatusNotSubmitted()}
+            </strong>
+          </span>
+
+          {/* This is a floating 'link' that resends the pending email. */}
+          {props.pendingEmail && (
+            <Button
+              styleAsText={true}
+              style={{...styles.resendLink, float: isRTL ? 'left' : 'right'}}
+              text={i18n.sessionLockoutResendEmail()}
+              onClick={resendPermissionEmail}
+            />
+          )}
+        </div>
+
+        {/* This field allows the input of an email address. */}
+        {/* Parent Email: [email] */}
+        <div style={styles.sections}>
+          <div style={styles.section}>
+            <label
+              style={isRTL ? styles.labelRTL : styles.label}
+              htmlFor="parent-email"
+            >
+              <strong>{i18n.sessionLockoutParentEmailField()}</strong>
+            </label>
+
+            {/* Slightly complicated layout allows for text underneath. */}
+            <div style={styles.fieldSection}>
+              <input
+                style={styles.field}
+                onChange={onEmailUpdate}
+                onInput={onEmailUpdate}
+                onBlur={onEmailUpdate}
+                defaultValue={props.pendingEmail}
+                id="parent-email"
+              />
+
+              {/* Show a 'Last email sent' prompt when available. */}
+              {props.pendingEmail && (
+                <p style={styles.lastEmail}>
+                  <em>
+                    Last email sent:{' '}
+                    {props.requestDate.toLocaleDateString(locale, {
+                      ...dateOptions,
+                      hour: 'numeric',
+                      minute: 'numeric',
+                    })}
+                  </em>
+                </p>
+              )}
+            </div>
+          </div>
+        </div>
+
+        {/* A sign-out button. */}
+        <Button
+          style={styles.button}
+          text={i18n.signOutButton()}
+          color={Button.ButtonColor.gray}
+          href="/users/sign_out"
+        />
+
+        {/* The submit button. */}
+        {/* An empty onClick will still submit the form. */}
+        <Button
+          style={{...styles.button, float: isRTL ? 'left' : 'right'}}
```

Comment:
Maybe a nit, but could we use flexbox to avoid using `float` and `isRTL` customization? Would the flexbox `justify-content: space-between` achieve the same spacing between the "Sign out" and "Send permission request" buttons?

Reply:
This is an amazing idea.

Reply:
That worked really well and now I want to revisit the past 10 years of my life and find all the places where I should have done this. Thank you!